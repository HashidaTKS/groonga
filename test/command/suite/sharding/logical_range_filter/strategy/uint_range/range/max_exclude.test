#@include fixture/sharding/logical_range_filter/strategy/uint_range/schema.grn

load --table Logs_2018030
[
{"year_day": 2018030, "memo": "2018-01-30 00:00:00", "message": "Start"},
{"year_day": 2018038, "memo": "2018-02-07 00:00:00", "message": "Shutdown"},
{"year_day": 2018039, "memo": "2018-02-08 00:00:00", "message": "Start"},
{"year_day": 2018039, "memo": "2018-02-08 23:59:59", "message": "Shutdown"}
]

load --table Logs_2018040
[
{"year_day": 2018040, "memo": "2018-02-09 00:00:00", "message": "Start"},
{"year_day": 2018048, "memo": "2018-02-17 00:00:00", "message": "Shutdown"},
{"year_day": 2018049, "memo": "2018-02-18 00:00:00", "message": "Start"},
{"year_day": 2018049, "memo": "2018-02-18 23:59:59", "message": "Shutdown"}
]

load --table Logs_2018050
[
{"year_day": 2018050, "memo": "2018-02-19 00:00:00", "message": "Start"},
{"year_day": 2018058, "memo": "2018-02-27 00:00:00", "message": "Shutdown"},
{"year_day": 2018059, "memo": "2018-02-28 00:00:00", "message": "Start"},
{"year_day": 2018059, "memo": "2018-02-28 23:59:59", "message": "Shutdown"}
]

logical_range_filter \
  --logical_table Logs \
  --shard_key year_day \
  --strategy "uint-range" \
  --filter 'message == "Shutdown"' \
  --max 2018049 \
  --max_border "exclude"

