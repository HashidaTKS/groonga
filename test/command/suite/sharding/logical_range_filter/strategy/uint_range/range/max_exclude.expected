load --table Logs_2018030
[
{"year_day": 2018030, "memo": "2018-01-30 00:00:00", "message": "Start"},
{"year_day": 2018038, "memo": "2018-02-07 00:00:00", "message": "Shutdown"},
{"year_day": 2018039, "memo": "2018-02-08 00:00:00", "message": "Start"},
{"year_day": 2018039, "memo": "2018-02-08 23:59:59", "message": "Shutdown"}
]
[[0,0.0,0.0],4]
load --table Logs_2018040
[
{"year_day": 2018040, "memo": "2018-02-09 00:00:00", "message": "Start"},
{"year_day": 2018048, "memo": "2018-02-17 00:00:00", "message": "Shutdown"},
{"year_day": 2018049, "memo": "2018-02-18 00:00:00", "message": "Start"},
{"year_day": 2018049, "memo": "2018-02-18 23:59:59", "message": "Shutdown"}
]
[[0,0.0,0.0],4]
load --table Logs_2018050
[
{"year_day": 2018050, "memo": "2018-02-19 00:00:00", "message": "Start"},
{"year_day": 2018058, "memo": "2018-02-27 00:00:00", "message": "Shutdown"},
{"year_day": 2018059, "memo": "2018-02-28 00:00:00", "message": "Start"},
{"year_day": 2018059, "memo": "2018-02-28 23:59:59", "message": "Shutdown"}
]
[[0,0.0,0.0],4]
logical_range_filter   --logical_table Logs   --shard_key year_day   --strategy "uint-range"   --filter 'message == "Shutdown"'   --max 2018049   --max_border "exclude"
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        "memo",
        "ShortText"
      ],
      [
        "message",
        "Text"
      ],
      [
        "year_day",
        "UInt32"
      ]
    ],
    [
      "2018-02-07 00:00:00",
      "Shutdown",
      2018038
    ],
    [
      "2018-02-08 23:59:59",
      "Shutdown",
      2018039
    ],
    [
      "2018-02-17 00:00:00",
      "Shutdown",
      2018048
    ]
  ]
]
